### DISCUSSION TITLE
Where to start debugging this? app fails to load properly at startup and requires system is rebooted to work

### DISCUSSION THREAD

**Discussion by lectrical** [keepassxreboot/keepassxc#11501] [Q&A]
Hello, i have had an ongoing Windows 11 x64 issue for some time, across various versions of Windows 11 and currently on  

![image](https://github.com/user-attachments/assets/1d8a96f1-0713-499f-90e0-6f6f32bf4cbd)

This is the problem.

When windows starts, if the taskbar icon does not load, then keepassxc will never work for me until i reboot the system. If the taskbar icon loads things are fine.

If the taskbar icon failed to load, the program will be running in task-manager, i can kill the process and start it. If i load a browser with the extensions the proxy exe loads but i cannot interact with the application in any meaningful way.

- The extensions cannot connect to the main db.
- I cannot open the ui of the app.
- There is no taskbar icon.
- logging out of session and back in does not fix it.
- reinstalling does not fix it.
- installing vs redistributable does not fix it.

I seem to have at a ratio of about 70% success at the minute and that is why i tolerated it for all this time. Recently it's been happening more often than not and it's starting to really bug me. I'll be in the middle of doing something and need to use keepassxc only to realize it didn't load this session and have to reboot.

I need to resolve this but don't know where to start to debug. I don't know if something else is doing this to keepassxc or the result of a bug in the program itself.

Reinstalling the app does not fix it. Installing vrcred does not fix it. Only restarting the system will fix it, potentially.

How do I go about debugging this issue please?

**Comment by droidmonkey** [comment]
Kill KeePassXC from the task manager. There should be no need to reboot. Delete `%localappdata%/KeePassXC/keepassxc.ini`. start keepassxc again. Exit. Reboot. See what happens.

**Reply by lectrical** [reply]
@droidmonkey thanks,

I'll try this next time I reboot and have the issue, then update here.

**Reply by lectrical** [reply]
@droidmonkey 

So it happened today, i did as advised and it did not solve the issue.

I deleted `%localappdata%/KeePassXC/keepassxc.ini` and restarted via task manager. After restart i just see this 

![image](https://github.com/user-attachments/assets/f59f6636-74dd-4fe4-9d85-ed5afaa706e0)

No new `%localappdata%/KeePassXC/keepassxc.ini` is created.

I also deleted `Roaming/KeePassXC/keepassxc.ini` and after restart is creates a new ini but just has this and nothing else.

```ini
[General]
ConfigVersion=2
```

So right now it is just stuck again. I've always assumed it was some kind of conflict with something i have installed, like docker desktop but never really sure how to debug it.

What can i try now?

**Reply by droidmonkey** [reply]
Unfortunately, I have no idea. A couple of other people ran into an issue like yours in the past. That was not resolved if I recall correctly because they got to this same state.

**Reply by droidmonkey** [reply]
See https://github.com/keepassxreboot/keepassxc/issues/4079 and https://github.com/keepassxreboot/keepassxc/issues/8685

**Reply by lectrical** [reply]
@droidmonkey

When it starts and works it has more memory usage.

![image](https://github.com/user-attachments/assets/0ed3fe2a-f08a-4a98-abc3-a068a2dbef7d)

Also it seems to do something with my usb adapters and yubikeys when exiting and loading, making the unmount/mount bell noise windows when It close/open it.

Could it be related to that. I never thought to remove the usb adapters because it don't use my yubikeys with keepass xc. I have a few of them active at startup though.

I already rebooted. I'll have try next time

Also, if i do `PS C:\Program Files\KeePassXC> .\KeePassXC.exe --debug-info` nothing happens. What is the expected result of that command?

**Reply by droidmonkey** [reply]
That flag is for qt

**Reply by lectrical** [reply]
@droidmonkey OK and thanks.

Next time it happens i'll look more into my yubikeys and peripherals to see if they are somehow related to the issue.

**Reply by lectrical** [reply]
So happened again today, and I have two usb hubs, with their own yubikeys.

Disconnected the first one, killed keepassxc via task manager and opened it, nothing.

Disconnected the second one, killed keepassxc via task manager and opened it, it opened as normal. Fully functional again.

So it seems to be something to do with that hub/yubikey. I figure maybe more likely the yubikey than the hub but I did not test removing the key before the hub. just unplugged the whole hub.

Two different brand hubs. First has v5 yuibikeys only. Second has a single v4 yubikey.

This is the first time I've been able to open keepassc after booting into this issue and this seems to have narrowed down the cause.

**Reply by droidmonkey** [reply]
Sounds like a lack of timeout or deadlock.  We do use mutexes with the yubikey code

**Reply by lectrical** [reply]
The only other context I forgot to mention is they are used with gpg4win.

Is there anything I can do to provide helpful information the next time it happens now i know where to look?

**Reply by droidmonkey** [reply]
It is quite possible we are racing gpg4win or another program for access to the key. We could be opening the key and during that time gpg4win takes over, closes the connection and then we are left in a wait condition. However yubikey is done on a thread separate from gui.

**Reply by lectrical** [reply]
I'll do more checks next time it happens to see if there is something reproducible.

In hindsight I do occasionally have some issues where gpg4win was not loading keys i had to restart kleopatra and the smartcard services to be able to use the keys to sign commits.

I had not connected the two issues as possibly related though so i will try to check if there if there is some relationship between the two programs where the issue next occurs.

**Reply by droidmonkey** [reply]
There is a problem on our end because keepassxc is not behaving in the presence of this issue, but it's hard to pin down where that failure is right now.

**Reply by lectrical** [reply]
OK, next time it happens i'll try killing off gpgwin and related services to see if that allows keepassxc to start without removing the usb hub.

I kind of know what i'm looking for now so i'll try to see if i can narrow the cause down further.

**Reply by droidmonkey** [reply]
Do you have more than 4 yubikeys conected to your computer?

**Reply by lectrical** [reply]
No, I have 3. 2 v5 via one hub and a v4 via another.

**Reply by lectrical** [reply]
(just an update) So happened again today. 

- Killed off gpg related services. nothing.
- usb devices on the hub were working and no device manager issues (on this hub)
- my two smartcards in the other hub had not loaded properly for reason, resolved by reinserting them.
- removing usb devices and reinserting them did nothing to allow the UI to load.
- Unplugging the hub itself worked to let keepassx load the UI. 

this usb hub is going through a usb c / thunderbolt case hub connector that came with my case.

It's appears to have something to do with the external hub or the case hub or both but they appear to function normally whilst keepassxc won't load the ui until it's disconnected.

I have decided to remove this expansion hub and use just one for all yubikeys. I'll see how it goes.

**Reply by droidmonkey** [reply]
Very interesting. I actually tested the yubikey connection code extensively using a USB hub and 3-4 keys before we merged that into the code base.
